common/shell-local/config.go

```diff
@@ -29,6 +29,9 @@ type Config struct {
 	// The shebang value used when running inline scripts.
 	InlineShebang string `mapstructure:"inline_shebang"`
 
+	// An array of multiple Runtime OSs to run on.
+	OnlyOn []string
```

Comment:
Generally we use underscores for our packer template, rather than camelcase.  So you'll want the name of the option you provide in your example to be `only_on` rather than `OnlyOn`, and the way you make that happen is by adding the mapstructure tags to this line:
```
OnlyOn                   string `mapstructure:"only_on"`
```

Reply:
I'd love to get a check in the [validate function](https://github.com/hashicorp/packer/blob/master/common/shell-local/config.go#L80) that checks that whatever user input was given matches something on the list of [valid GOOS values](https://github.com/golang/go/blob/bf8e6b70276ddfd3e070964542020fe782c1eec8/src/go/build/syslist.go#L7); this will prevent painful user errors from happening because of typos.

Reply:
yeah mapstructure is a bit of magic 😅

Reply:
Added the `mapstructure` piece and updated the docs for both provisioner and post-processor.

Added an initial stab at validate. Think it can be simplified but it should work. :)